penguin/tools/browser_harness_ownership.py

```
@@ -0,0 +1,113 @@
+            for name, record in records.items()
+            if isinstance(record, dict)
+        }
+
+    def is_owned(self, bu_name: str) -> bool:
+        record = self.get(bu_name)
+        return bool(record and record.get("started_by_penguin"))
+
+    def record_started(self, identity: Dict[str, Any]) -> Dict[str, Any]:
+        bu_name = str(identity.get("bu_name") or "")
+        if not bu_name:
+            raise ValueError("Cannot persist browser ownership without bu_name")
```

Ruff (0.15.12)

[warning] Avoid specifying long messages outside the exception class

(TRY003)